### PR TITLE
Fix several warnings exposed in MSVS.

### PR DESCRIPTION
**Pull request by null77**
diagnostic.cpp:
- unreachable code

operand.cpp
- conversion between int and uint32_t
- unreachable code

text.cpp:
- assignment within conditional expression

hex_float.h:
- assignment within conditional expression
- conversion from 'const int' to 'unsigned int'
- unreachable code

validate_id.cpp
- forcing value to bool 'true' or 'false'

validate_types.cpp:
- forcing value to bool 'true' or 'false'

### CODE REVIEW

File: source/util/hex_float.h [hunk 1]
```diff
@@ -662,12 +659,19 @@ inline uint8_t get_nibble_from_character(int character) {
   const char* dec = "0123456789";
   const char* lower = "abcdef";
   const char* upper = "ABCDEF";
-  const char* p = nullptr;
-  if ((p = strchr(dec, character))) {
+  const char* p = p = strchr(dec, character);
```
Comment (dekimir) [inline source/util/hex_float.h]: Does 'p =' really have to be repeated?

**Comment by dekimir** [comment]
BTW, which MSVS version is this?  We seem to have no trouble on 2013 and 2015 with the code as it currently stands.

Also, I would oppose a blanket ban on assignment within conditional expression going forward.

File: source/util/hex_float.h (hunk 1 above)
Comment (dekimir) [inline source/util/hex_float.h]: Interestingly, this line seems to break the gcc build (see "checks have failed" below).

**Comment by null77** [comment]
Updated CL, reverted the assignment within conditional fix. The extra 'p' was a bug, and it's cool that the CI caught that. Updated commit message to show that I'm using VS2015.

**Comment by dekimir** [comment]
Rebased and merged to master via 34cb003.

**Pull request closed without merging**